### PR TITLE
erlang@21 21.3.8.18

### PR DESCRIPTION
**Pull request by gromgit**
Created with `brew bump-formula-pr`.

~~`resource` blocks may require updates.~~ Documentation tarballs haven't changed.

### CODE REVIEW

**Comment by bayandin** [comment]
```
drivers/common/inet_drv.c:1480:30: error: static declaration of 'in6addr_any' follows non-static declaration
static const struct in6_addr in6addr_any = { { IN6ADDR_ANY_INIT } };
                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/netinet6/in6.h:211:30: note: previous declaration is here
extern const struct in6_addr in6addr_any;
                             ^
drivers/common/inet_drv.c:1489:30: error: static declaration of 'in6addr_loopback' follows non-static declaration
static const struct in6_addr in6addr_loopback =
                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/netinet6/in6.h:212:30: note: previous declaration is here
extern const struct in6_addr in6addr_loopback;
```

We could try to cherry-pick this commit https://github.com/erlang/otp/commit/388622e9b626039c1e403b4952c2c905af364a96 (https://bugs.erlang.org/browse/ERL-1306)

File: Formula/erlang@21.rb [hunk 1]
```diff
@@ -40,12 +40,6 @@ class ErlangAT21 < Formula
...
   end
 
-  # Fix build on Xcode 11.4
-  patch do
-    url "https://github.com/erlang/otp/commit/3edba0dad391431cbadad44a8bd15c75254fc239.patch?full_index=1"
-    sha256 "0c82d9f3bdb668ba78025988c9447bebe91a2f6bb00daa7f0ae7bd1916cd9bfd"
-  end
-
   def install
     # Work around Xcode 11 clang bug
```
Comment (bayandin) [inline Formula/erlang@21.rb:44]: We can delete this workaround as well

Comment (chenrui333) [reply Formula/erlang@21.rb:44]: 👍

**Comment by jonchang** [comment]
Catalina failure is likely related to the Xcode implicit function declaration change.

> Clang now reports an error when you use a function without an explicit declaration when building C or Objective-C code for macOS (-Werror=implicit-function-declaration flag is on). This additional error detection unifies Clang’s behavior for iOS/tvOS and macOS 64-bit targets for this diagnostic. (49917738)

**Comment by jonchang** [comment]
> We could try to cherry-pick this commit erlang/otp@388622e (https://bugs.erlang.org/browse/ERL-1306)

Yeah, let's do this.

**Comment by gromgit** [comment]
> We could try to cherry-pick this commit [erlang/otp@388622e](https://github.com/erlang/otp/commit/388622e9b626039c1e403b4952c2c905af364a96) (https://bugs.erlang.org/browse/ERL-1306)

Thanks @bayandin, trying it out now.

**Comment by gromgit** [comment]
Looks like the previous patch is still required, reinstating...

**Comment by chenrui333** [comment]
let's squash the commits and go with the bump. Do we have an upstream issue tracker for this?

**Comment by gromgit** [comment]
Commits squashed.

> Do we have an upstream issue tracker for this?

Not sure what you mean. I've added the links to the two upstream tickets for future reference.

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**